File: csharp/07-switch-factory/src/playlist_curator.cs
public class Track
{
    public string Title = "";
    public int Tempo;
    public int Energy;
    public bool Instrumental;
    public int Danceability;
}

public class PlaylistCurator
{
    public List<Track> create_playlist(string mood, List<Track> tracks)
    {
        var result = new List<Track>();
        switch (mood)
        {
            case "happy":
                result = tracks.Where(t => t.Tempo > 120)
                              .OrderByDescending(t => t.Tempo)
                              .Take(20)
                              .ToList();
                break;

            case "sad":
                result = tracks.Where(t => t.Tempo < 90)
                              .OrderBy(t => t.Tempo)
                              .Take(15)
                              .ToList();
                break;

            case "workout":
                result = tracks.Where(t => t.Tempo > 130 && t.Energy > 7)
                              .Take(25)
                              .ToList();
                break;

            case "focus":
                result = tracks.Where(t => t.Instrumental)
                              .Take(20)
                              .ToList();
                break;

            case "party":
                result = tracks.Where(t => t.Tempo >= 120 && t.Danceability >= 6)
                              .Take(30)
                              .ToList();
                break;

            default:
                result = tracks.Take(10).ToList();
                break;
        }
        return result;
    }
}

File: python/07-switch-factory/src/playlist_curator.py
class PlaylistCurator:
    def __init__(self):
        pass

    def create_playlist(self, mood, tracks):
        result = []

        if mood == "happy":
            for track in tracks:
                if track.get("tempo", 0) > 120:
                    result.append(track)
            result.sort(key=lambda x: x.get("tempo", 0), reverse=True)
            result = result[:20]

        elif mood == "sad":
            for track in tracks:
                if track.get("tempo", 0) < 90:
                    result.append(track)
            result.sort(key=lambda x: x.get("tempo", 0))
            result = result[:15]

        elif mood == "workout":
            for track in tracks:
                if track.get("tempo", 0) > 130 and track.get("energy", 0) > 7:
                    result.append(track)
            result.sort(key=lambda x: x.get("energy", 0), reverse=True)
            result = result[:25]

        elif mood == "focus":
            for track in tracks:
                if track.get("instrumental") is True:
                    result.append(track)
            result.sort(key=lambda x: x.get("tempo", 0))
            result = result[:30]

        elif mood == "party":
            for track in tracks:
                if track.get("tempo", 0) > 110 and track.get("danceability", 0) > 6:
                    result.append(track)
            result.sort(key=lambda x: x.get("danceability", 0), reverse=True)
            result = result[:20]

        else:
            for track in tracks:
                result.append(track)
            result = result[:10]

        return result
